From 8572487f1159bd4af667800dffcc11c44bbfbab0 Mon Sep 17 00:00:00 2001
From: Richard Guy Briggs <rgb@redhat.com>
Date: Fri, 19 Dec 2014 00:02:06 -0500
Subject: [powerpc] simplify syscall_get_arch()

Message-id: <eadcedf32221a8bcb09755377939d8ffacccdc64.1418931685.git.rgb@redhat.com>
Patchwork-id: 102326
O-Subject: [RHEL7 PATCH 1/2] powerpc: simplify syscall_get_arch()
Bugzilla: 1169461
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Paul Moore <pmoore@redhat.com>
RH-Acked-by: Kyle McMartin <kmcmarti@redhat.com>

From: Eric Paris <eparis@redhat.com>

Was erroneously called upstream: sparc: simplify syscall_get_arch()

Include linux/thread_info.h so we can use is_32_bit_task() cleanly.
Then just simplify syscall_get_arch() since is_32_bit_task() works for
all configuration options.

Upstream commit: 75dddcbd

[RGB: Dropped the header file changes since it caused a not strictly necessary
kABI change.]

Suggested-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Eric Paris <eparis@redhat.com>
Signed-off-by: Richard Guy Briggs <rgb@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/syscall.h b/arch/powerpc/include/asm/syscall.h
index 4271544..9caf413 100644
--- a/arch/powerpc/include/asm/syscall.h
+++ b/arch/powerpc/include/asm/syscall.h
@@ -90,12 +90,6 @@ static inline void syscall_set_arguments(struct task_struct *task,
 
 static inline int syscall_get_arch(void)
 {
- int arch = AUDIT_ARCH_PPC;
-
-#ifdef CONFIG_PPC64
- if (!is_32bit_task())
-  arch = AUDIT_ARCH_PPC64;
-#endif
- return arch;
+ return is_32bit_task() ? AUDIT_ARCH_PPC : AUDIT_ARCH_PPC64;
 }
 #endif /* _ASM_SYSCALL_H */
-- 
1.7.1